core/runner.py

```diff
@@ -81,6 +80,21 @@ def run_job(self, job):
         logging.info("Finished running job! (type: {0})".format(job.job_type))
         return results
 
+    def _set_up(self, job):
```

Comment:
Doing `count_datapoints()` on every job seems expensive. Could we do this once during `_transform_and_split_data()` and return it in the results?